ServerCore/SyncController.cs

```diff
@@ -489,8 +484,74 @@ public SyncController(PuzzleServerContext context, UserManager<IdentityUser> use
 
             var helper = new SyncHelper(context);
             var response = await helper.GetSyncResponse(currentEvent.ID, team.ID, puzzleId, query_puzzle_ids, min_solve_count,
-                                                        annotations, last_sync_time);
+                                                        annotations, last_sync_time, false);
             return Json(response);
         }
+        
+        // GET api/Sync/client
+        [HttpGet]
+        [Route("{eventId}/{puzzleId}/api/Sync/client")]
+        public async Task<IActionResult> Index(string eventId, int puzzleId)
+        {
+            Event currentEvent = await EventHelper.GetEventFromEventId(context, eventId);
+            if (currentEvent == null)
+            {
+                return Content("ERROR:  That event doesn't exist");
+            }
+            
+            PuzzleUser user = await PuzzleUser.GetPuzzleUserForCurrentUser(context, User, userManager);
+            if (user == null)
+            {
+                return Content("ERROR:  You aren't logged in");
+            }
+
+            Team team = await UserEventHelper.GetTeamForPlayer(context, currentEvent, user);
+            if (team == null)
+            {
+                return Content("ERROR:  You're not on a team");
+            }
+
+            Puzzle thisPuzzle = await context.Puzzles.FirstOrDefaultAsync(m => m.ID == puzzleId);
+            if (thisPuzzle == null)
+            {
+                return Content("ERROR:  That's not a valid puzzle ID");
+            }
+
+            // Start doing a sync asynchronously while we download the file contents.
+
+            var helper = new SyncHelper(context);
+            Task<Dictionary<string, object>> responseTask = helper.GetSyncResponse(currentEvent.ID, team.ID, puzzleId, null, 0, null, null, true);
+
+            // Find the material file with the latest-alphabetically ShortName
+
+            var materialFiles = await (from f in context.ContentFiles
```

Comment:
Why not just delete the old material file so there's always one anyway? If that's a bad idea, why not use OrderByDescending here and take the first one?

Reply:
The main reason for having code dealing with multiple concurrent versions is that it's currently not supported to overwrite an existing materials file. :-(  So if you want to make sure there isn't a point where there's no materials file, you have to first add one and then delete the old one.  So there's a point where there are more than one, and I wanted code to handle that.

And, yes, it's a good idea to switch to use OrderByDescending.  I'll do that.